### ISSUE TITLE
Can't call conditional expression

### ISSUE THREAD

**Issue by andy-hanson** [AssemblyScript/assemblyscript#255]
```ts
function f(x: i32): i32 { return x; }
function g(x: i32): i32 { return x + 2; }
export function test(b: boolean): i32 {
  return (b ? f : g)(1);
}
```

Currently fails with: 
```
ERROR TS1005: ')' expected.

   return (b ? f : g)(1);
             ~
```

**Comment by MaxGraey** [comment]
Hmm, for this specific case when signature are identical it pretty simple, we can just syntactic sugaring to `b ? f(1) : g(1)`, but that about this:
```ts
function f(x: i32): i32 { return x; }
function g(x: i32, y: i32): i32 { return x + 2; }

function test(b: boolean): i32 {
  return (b ? f : g)(1, 2);
}
```
This valid in Typescript btw

**Comment by andy-hanson** [comment]
This is just a parser error -- you can already do this in master using `let fn = b ? f : g;`. It looks like AssemblyScript doesn't do function subtyping, so it's an error if `f` and `g` aren't exactly the same type.

**Comment by dcodeIO** [comment]
Yeah, it can't resolve to a concrete `Function` in this case, but there's also something I simply called `FunctionTarget` that should work, that is if the resolver had code to combine a left and right `Function`s to a `FunctionTarget`.

**Comment by MaxGraey** [comment]
Resolved in  #833
